### PR TITLE
Parse the normalized container.PortMappings presented by the Marathon 1.5.x API

### LINKED ISSUE #3465: Marathon Service Discovery `portMappings` meta label broken for Marathon 1.5.x
**What did you do?**
Started prometheus with a marathon service discovery configuration that uses the `__meta_marathon_port_mapping_label_<labelname>` meta tag in a `relabel_config` to determine whether or not to keep the matching port for scraping and at which path to scrape:

**What did you expect to see?**
Targets populated within 1 minute of boot

**What did you see instead? Under which circumstances?**
No targets populated

**Environment**
CoreOS 1235.12.0
Amazon AWS EC2

* System information:
Linux 4.13.9-coreos x86_64

* Prometheus version:
Tested on multiple versions:

```
prometheus, version 2.0.0 (branch: HEAD, revision: 0a74f98628a0463dddc90528220c94de5032d1a0)
  build user:       root@615b82cb36b6
  build date:       20171108-07:11:59
  go version:       go1.9.2
```

```
prometheus, version 1.7.1 (branch: master, revision: 3afb3fffa3a29c3de865e1172fb740442e9d0133)
  build user:       root@0aa1b7fc430d
  build date:       20170612-11:44:05
  go version:       go1.8.3
```

* Prometheus configuration file:
```
global:
  scrape_interval: 15s

scrape_configs:
  - job_name: 'marathon'
    scrape_interval: '5s'
    marathon_sd_configs:
      - servers: ['http://marathon-service:8080']
    relabel_configs:
      # disable all targets which do not have a METRICS_PATH on them
      - source_labels: [__meta_marathon_port_mapping_label_METRICS_PATH]
        action: keep
        regex: (.+)

      # assign the value of METRICS_PATH to the prometheus target's metrics path
      - source_labels: [__meta_marathon_port_mapping_label_METRICS_PATH]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
```

* Logs:
```
level=info ts=2017-11-13T17:24:47.263849987Z caller=main.go:215 msg="Starting Prometheus" version="(version=2.0.0, branch=HEAD, revision=0a74f98628a0463dddc90528220c94de5032d1a0)"
level=info ts=2017-11-13T17:24:47.263914012Z caller=main.go:216 build_context="(go=go1.9.2, user=root@615b82cb36b6, date=20171108-07:11:59)"
level=info ts=2017-11-13T17:24:47.263949629Z caller=main.go:217 host_details="(Linux 4.13.9-coreos #1 SMP Thu Oct 26 03:21:00 UTC 2017 x86_64 ip-10-0-15-95.ec2.internal (none))"
level=info ts=2017-11-13T17:24:47.267971677Z caller=web.go:380 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2017-11-13T17:24:47.267938359Z caller=main.go:314 msg="Starting TSDB"
level=info ts=2017-11-13T17:24:47.270025324Z caller=targetmanager.go:71 component="target manager" msg="Starting target manager..."
level=info ts=2017-11-13T17:24:47.877861771Z caller=main.go:326 msg="TSDB started"
level=info ts=2017-11-13T17:24:47.877924302Z caller=main.go:394 msg="Loading configuration file" filename=/etc/prometheus/prometheus.yml
level=info ts=2017-11-13T17:24:47.878519327Z caller=main.go:371 msg="Server is ready to receive requests."
```

### PR DESCRIPTION
**Pull request by whoward**
Fixes #3465 (see this for a full description for the error I'm trying to fix)

@fabxc I think you would be the most appropriate maintainer for this PR

Please feel free to let me know about styling issues - I'm definitely a newbie when it comes to golang so there's a good chance I've stumbled over some 'no-no's while writing this code.

### CODE REVIEW

File: discovery/marathon/marathon.go [hunk 1]
```diff
@@ -350,6 +351,12 @@ func targetsForApp(app *App) []model.LabelSet {
 					target[model.LabelName(ln)] = model.LabelValue(lv)
 				}
 			}
+			if i < len(app.Container.PortMappings) {
```
Comment (Conorbro) [inline discovery/marathon/marathon.go]: Could you add comments indicating this was added to accommodate for the change you noticed with Marthon 1.5.x and another to the previous section indicating that it's for Marthon versions <=  Marathon 1.4.x ?

Comment (whoward) [reply discovery/marathon/marathon.go]: sure - added some comments - let me know if that works!

Comment (Conorbro) [reply discovery/marathon/marathon.go]: Perfect. I'll merge once the CircleCI tests pass again. Thanks for fixing!

File: discovery/marathon/marathon.go [hunk 2]
```diff
@@ -222,7 +222,8 @@ type DockerContainer struct {
 
 // Container describes the runtime an app in running in.
 type Container struct {
-	Docker DockerContainer `json:"docker"`
+	Docker       DockerContainer `json:"docker"`
+	PortMappings []PortMappings  `json:"portMappings"` // present in Marathon 1.5.x
```
Comment (brian-brazil) [inline discovery/marathon/marathon.go]: Comments should start with a capital letter and end in a full stop. I'd remove this comment, as the below ones cover this.

Comment (whoward) [reply discovery/marathon/marathon.go]: updated

**Comment by whoward** [comment]
thanks! looking forward to the next release!

**Pull request merged**